src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts

```diff
@@ -560,11 +562,18 @@ export class NotebookService extends Disposable implements INotebookService {
 		if (notebookUri.scheme === Schemas.untitled) {
 			return true;
 		}
+		const trustedBooksConfigKey = 'notebook.trustedBooks';
 
 		let cacheInfo = this.trustedNotebooksMemento.trustedNotebooksCache[notebookUri.toString()];
 		if (!cacheInfo) {
-			// This notebook was never trusted
-			return false;
+			// Check if the notebook belongs to a book that's trusted.
+			let trustedBookDirectories: string[] = this._configurationService?.getValue(trustedBooksConfigKey) ?? [];
+			if (trustedBookDirectories.find(b => notebookUri.fsPath.indexOf(b) > -1)) {
```

Comment:
How are we handling untrusting a notebook that's in a trusted book? Should we even allow that?

Reply:
Probably when a book that's trusted is not trusted, it goes into the untrustedQueue list. @joberume is the expert here. Can you correct me if am wrong?